### DISCUSSION TITLE
Exploding of new data types

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#101] [General]
What do the *Holder data types serve? I guess they are the equivalent to pointers? e.g: IntHolder = Int pointer

https://github.com/objeck/objeck-lang/commit/4ecd2a5bcccb246e76f08ed75d723064c03936ec#diff-ff7e7431575af6bfe6f89f2268e7abdeb170e1f9825e1bc09b91eff2d37d81ed

**Comment by objeck** [comment]
The Holder classes are references to primitives. For example `-13->Abs()->PrintLine()` is valid code. If I want to add pass -13 as an object instance, an `IntHolder ` is used. Objeck supports boxing and unboxing, so `vector->AddBack(-13);` is valid syntax as the compiler converts the code to `vector->AddBack(IntHolder->New(-13));`. 

This abstraction is lost when passing object instances to C functions. A lightweight [C++ API](https://github.com/objeck/objeck-lang/blob/master/core/vm/lib_api.h) is used to pass objects back and forth between the Objeck VM and 3rd party C-shared libraries.

**Reply by ghost** [reply]
So they are like Java's Integer, Float,... wrapping primitive types?

**Comment by objeck** [comment]
Yes
